### PR TITLE
Add a dedicated page for the recommended apps installation

### PR DESCRIPTION
**Pull request by ChristophWurst**
Todo
- [x] redirect to this page on setup submit @ChristophWurst 
- [x] app fetch error handling @ChristophWurst 
- [x] installation error handling @ChristophWurst 
- [x] redirect when done @ChristophWurst 
- [x] Make pretty @jancborchardt 

---

Current state

Bildschirmfoto von 2019-12-05 11-28-11

<details>
old versions
Bildschirmfoto von 2019-12-05 10-33-15
Bildschirmfoto von 2019-12-05 10-03-12
Bildschirmfoto von 2019-12-04 19-58-06
</details>

### CODE REVIEW

File: core/src/recommendedapps.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { translate as t } from '@nextcloud/l10n'
+import Vue from 'vue'
+
+import logger from './logger'
+import RecommendedApps from './components/setup/RecommendedApps'
+
+// eslint-disable-next-line camelcase
+__webpack_nonce__ = btoa(getRequestToken())
+// eslint-disable-next-line camelcase
+__webpack_public_path__ = generateFilePath('mail', '', 'js/')
```
Comment (jancborchardt) [inline core/src/recommendedapps.js]: :>

**Review by rullzer — APPROVED** [review]
Does the trick and already looks a lot better!

File: apps/settings/src/components/AppList.vue [hunk 2]
```diff
@@ -131,26 +129,26 @@ export default {
 					return OC.Util.naturalSortCompare(sortStringA, sortStringB)
 				})
 
-			switch (this.category) {
-			case 'installed':
+			if (this.category === 'installed') {
```
Comment (skjnldsv) [inline apps/settings/src/components/AppList.vue]: much cleaner :p

File: core/Controller/RecommendedAppsController.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+	/**
+	 * @NoCSRFRequired
+	 * @return Response
+	 */
+	public function index(): Response {
+		$this->initialStateService->provideInitialState('core', 'defaultPageUrl', \OC_Util::getDefaultPageUrl());
+		$response = new StandaloneTemplateResponse($this->appName, 'recommendedapps', [], 'guest');
+		$csp = new ContentSecurityPolicy();
+		$csp->addAllowedImageDomain('https://usercontent.apps.nextcloud.com');
```
Comment (skjnldsv) [inline core/Controller/RecommendedAppsController.php]: Interesting!

Comment (rullzer) [reply core/Controller/RecommendedAppsController.php]: Ah I think that is a leftover from before

Comment (rullzer) [reply core/Controller/RecommendedAppsController.php]: yep. removed

**Comment by jancborchardt** [comment]
I will only have time to look into this in the next days. If someone else wants to have a quick look:
- [ ] App title should be white (same as text color)
- [ ] App icon should be white (same as text color)
- [ ] App description text should be left-aligned
- [ ] Error should just be bold (in a `<strong>` element) instead of in a `class="error"` container
- [ ] The width of the dialog should be fixed to what we had before. See **Container on update screen is too wide #17556**
- [ ] The "Go back" link is a bit confusing? Should it be "Cancel"?

**Pull request merged**